prime-dotnet-webapi/HttpClients/Email/CHESClient.cs

```diff
@@ -35,30 +40,61 @@ public async Task SendAsync(string from, IEnumerable<string> to, IEnumerable<str
             var requestParams = new ChesEmailRequestParams(from, to, subject, body, chesAttachments);
 
             var requestContent = new StringContent(JsonConvert.SerializeObject(requestParams), Encoding.UTF8, "application/json");
+            try
+            {
+                HttpResponseMessage response = await _client.PostAsync("email", requestContent);
+                var responseJsonString = await response.Content.ReadAsStringAsync();
 
-            HttpResponseMessage response = null;
+                if (response.IsSuccessStatusCode)
+                {
+                    var successResponse = JsonConvert.DeserializeObject<EmailSuccessResponse>(responseJsonString);
+                    return successResponse.Messages[0].MsgId;
+                }
+                else
+                {
+                    _logger.LogError($"CHES Response code: {(int)response.StatusCode}, response body:{responseJsonString}");
+                    return Guid.Empty;
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError($"CHES Exception: {ex.Message}");
+                throw new Exception("Error occurred when calling CHES Email API. Try again later.", ex);
+            }
+        }
+
+        public async Task<string> GetStatusAsync(Guid msgId)
+        {
             try
             {
-                response = await _client.PostAsync("email", requestContent);
+                HttpResponseMessage response = await _client.GetAsync($"status?msgId={msgId}");
+                var responseString = await response.Content.ReadAsStringAsync();
 
                 if (response.IsSuccessStatusCode)
                 {
-                    var responseJsonString = await response.Content.ReadAsStringAsync();
-                    JsonConvert.DeserializeObject<EmailSuccessResponse>(responseJsonString);
+                    var statusResponse = JsonConvert.DeserializeObject<IList<StatusResponse>>(responseString);
+                    return statusResponse[0].Status;
```

Comment:
If we expect this to only ever return a single status response, it would be more correct to use `statusResponse.Single().Status` instead. If there is maybe more than one but we only care about the first, then we can use `statusResponse.First().Status`